pkg/apis/cluster/v1alpha2/cluster_types.go

```diff
@@ -106,29 +90,53 @@ type NetworkRanges struct {
 type ClusterStatus struct {
 	// APIEndpoint represents the endpoint to communicate with the IP.
 	// +optional
-	APIEndpoints []APIEndpoint `json:"apiEndpoints,omitempty"`
-
-	// NB: Eventually we will redefine ErrorReason as ClusterStatusError once the
-	// following issue is fixed.
-	// https://github.com/kubernetes-incubator/apiserver-builder/issues/176
+	APIEndpoint APIEndpoint `json:"apiEndpoint,omitempty"`
 
 	// If set, indicates that there is a problem reconciling the
 	// state, and will be set to a token value suitable for
 	// programmatic interpretation.
 	// +optional
-	ErrorReason common.ClusterStatusError `json:"errorReason,omitempty"`
+	ErrorReason *common.ClusterStatusError `json:"errorReason,omitempty"`
 
 	// If set, indicates that there is a problem reconciling the
 	// state, and will be set to a descriptive error message.
 	// +optional
-	ErrorMessage string `json:"errorMessage,omitempty"`
+	ErrorMessage *string `json:"errorMessage,omitempty"`
+
+	// Phase represents the current phase of cluster actuation.
+	// E.g. Pending, Running, Terminating, Failed etc.
+	// +optional
+	Phase *string `json:"phase,omitempty"`
```

Comment:
Sorry to come back to pointer vs not, but is there specific value in being able to distinguish between `nil` and `""` here?

Reply:
Either we change all of them, or we keep it consistent for now. We can follow-up later. I don't mind either way

Reply:
It's fine like this for now. I'll file an issue to switch phase to a non-pointer.